### PR TITLE
Migrate rtl test case to it's playwright version

### PR DESCRIPTION
**Pull request by pavanpatil1**
## What?

Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ rtl-test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/rtl.test.js) to its Playwright version.
## Why?

Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/various/rtl.spec.js`

### CODE REVIEW

**Comment by pavanpatil1** [comment]
Hi @kevin940726, Hope you're doing well. Could you please help me with reviewing this PR?. 

All test cases are passed. But seems like one End-to-End Tests / Admin - 3 CI check is failing. The error message which is shown is not related to test case which I worked upon. the error is on the test case which I haven't modified. Can you take a look at it?

image

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 1]
```diff
@@ -41,21 +39,38 @@ describe( 'RTL', () => {
 
 		// Expect: ARABIC_ZERO + ARABIC_ONE + ARABIC_TWO (<p>٠١٢</p>).
 		// N.b.: HTML is LTR, so direction will be reversed!
-		expect( await getEditedPostContent() ).toMatchSnapshot();
+
+		//Check the content
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]: Nit: 
```suggestion
		// Check the content.
```

and ditto to the others below.

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 2]
```diff
@@ -77,51 +98,75 @@ describe( 'RTL', () => {
...
+		page,
+		pageUtils,
+	} ) => {
 		await page.keyboard.press( 'Enter' );
 
 		await page.keyboard.type( ARABIC_ZERO );
 		await page.keyboard.press( 'Enter' );
 		await page.keyboard.type( ARABIC_ONE );
-		await pressKeyWithModifier( 'shift', 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]: Nit: Maybe just use the vanilla `press` here for simplicity?

```suggestion
		await page.keyboard.press( 'Shift+Enter' );
```

Ditto to the others below.

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 3]
```diff
@@ -77,51 +98,75 @@ describe( 'RTL', () => {
...
-	it( 'should navigate inline boundaries', async () => {
+	test( 'should navigate inline boundaries', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
 		await page.keyboard.press( 'Enter' );
 
 		// Wait for rich text editor to load.
 		await page.waitForSelector( '.block-editor-rich-text__editable' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]: Could we change this selector to something accessible? For instance, selecting it using a role selector, or replacing it along with the `press( 'Enter' )` above with `page.click( 'role=button[name="Add default block"i]' )`.

**Review by kevin940726 — COMMENTED** [review]
Nice job! Mostly looks good to me 👍 

Yeah, I think the e2e failure is unrelated to this PR.

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 4]
```diff
@@ -77,51 +98,75 @@ describe( 'RTL', () => {
...
+		await page.keyboard.press( 'Shift+Enter' );
 		await page.keyboard.type( ARABIC_TWO );
 		await page.keyboard.press( 'ArrowRight' );
 		await page.keyboard.press( 'ArrowRight' );
 		await page.keyboard.press( 'ArrowRight' );
 
 		// Move to the previous block with two lines in the current block.
 		await page.keyboard.press( 'ArrowRight' );
-		await pressKeyWithModifier( 'shift', 'Enter' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]: Let's update this too for consistency.

```suggestion
		await page.press( 'Shift+Enter' );
```

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 5]
```diff
@@ -77,51 +98,75 @@ describe( 'RTL', () => {
...
 		await page.keyboard.type( ARABIC_ONE );
 
 		// Move to the next block with two lines in the current block.
 		await page.keyboard.press( 'ArrowLeft' );
 		await page.keyboard.type( ARABIC_ZERO );
-		await pressKeyWithModifier( 'shift', 'Enter' );
-
-		expect( await getEditedPostContent() ).toMatchSnapshot();
+		await pageUtils.pressKeyWithModifier( 'shift', 'Enter' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]: Ditto.

```suggestion
		await page.press( 'Shift+Enter' );
```

**Comment by pavanpatil1** [comment]
Hi @kevin940726, I have addressed all the above feedback. Could you please review it again ?. 

Thanks!.

**Comment by kevin940726** [comment]
Have you seen [this comment](https://github.com/WordPress/gutenberg/pull/41495#discussion_r889871590) and my [latest review](https://github.com/WordPress/gutenberg/pull/41495#pullrequestreview-1008625519)? :)

This is very close to being merged! Thanks for the hard work!

**Comment by pavanpatil1** [comment]
Hi @kevin940726, 

https://github.com/WordPress/gutenberg/pull/41495#discussion_r889871590 comment is solved here https://github.com/WordPress/gutenberg/blob/09e40c9fd269154b61ad1bf8ee850ce4719bf987/test/e2e/specs/editor/various/rtl.spec.js#L154. It is not showing updated due to the selected string was different. 

Could you please review it again?..

File: test/e2e/specs/editor/various/rtl.spec.js [hunk 6]
```diff
@@ -77,51 +98,74 @@ describe( 'RTL', () => {
...
 	} );
 
-	it( 'should navigate inline boundaries', async () => {
-		await page.keyboard.press( 'Enter' );
+	test( 'should navigate inline boundaries', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		page.click( 'role=button[name="Add default block"i]' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/rtl.spec.js]:
```suggestion
		await page.click( 'role=button[name="Add default block"i]' );
```

File: test/e2e/specs/editor/various/rtl.spec.js (hunk 3 above)
Comment (kevin940726) [reply test/e2e/specs/editor/various/rtl.spec.js]: We still need to refactor this line to use role selectors instead though. Or maybe just remove this line if it's not doing anything meaningful now?

**Comment by pavanpatil1** [comment]
Hi @kevin940726, Above feedbacks are addressed. Could you check once ?.

**Comment by kevin940726** [comment]
Have you seen https://github.com/WordPress/gutenberg/pull/41495#discussion_r901311068 ? GitHub makes it less discoverable 😅

File: test/e2e/specs/editor/various/rtl.spec.js (hunk 3 above)
Comment (pavanpatil1) [reply test/e2e/specs/editor/various/rtl.spec.js]: Hi @kevin940726, We have refactored that line and Added the role selector https://github.com/WordPress/gutenberg/blob/09e40c9fd269154b61ad1bf8ee850ce4719bf987/test/e2e/specs/editor/various/rtl.spec.js#L154. We can't remove that line I guess.

Comment (kevin940726) [reply test/e2e/specs/editor/various/rtl.spec.js]: I mean to remove this line.

```js
await page.waitForSelector( '.block-editor-rich-text__editable' );
```

I don't think we need it anymore?

**Comment by pavanpatil1** [comment]
Hi @kevin940726, All feedbacks are addressed could you review it again ?.

**Review by kevin940726 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**